exercises/concept/card-tricks/card_tricks_test.go

```diff
@@ -247,13 +246,32 @@ func TestRemoveItem(t *testing.T) {
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			if got := RemoveItem(copySlice(tt.args.slice), tt.args.index); !reflect.DeepEqual(got, tt.want) {
+			if got := RemoveItem(copySlice(tt.args.slice), tt.args.index); !slicesEqual(got, tt.want) {
 				t.Errorf("RemoveItem(slice:%v, index:%v) = %v, want %v", tt.args.slice, tt.args.index, got, tt.want)
 			}
 		})
 	}
 }
 
+func slicesEqual(a, b []int) bool {
+	if len(a) != len(b) {
+		return false
+	}
+
+	if len(a) == 0 {
+		return true
+	}
+
+	size := len(a)
+	for i := 0; i < size; i++ {
+		if a[i] != b[i] {
+			return false
+		}
+	}
+
+	return true
```

Comment:
nit: this is precisely the sort of code that is easy to have off-by-one errors on and would probably need to be tested on its own to be truly confident. Personally, I would trust this more if this part where delegated to reflect.DeepEquals since you already handled the `nil` case before getting here.

Reply:
A middle ground between making this code less susceptible to off-by-one errors while still not using the `reflect` package would be to do something like:

```go
for i := range a {
    if a[i] != b[i] {
        return false
    }
}
```

I can do a follow-up PR if you think it's worth it. Personally I also would like to not use the `reflect` package. In Exercism we kind of encourage students to look at the tests (specially in practice exercises), and introducing them to the reflect package via the tests seems overkill. Not only because we are showing them a package they should think twice before using, but also because we can write a simple function ourselves. By not using the reflect package we also avoid all the runtime checks for types that the reflect package brings.